### ISSUE TITLE
panic: runtime error: slice bounds out of range

### ISSUE THREAD

**Issue by calmh** [syncthing/syncthing#47]
```
[YUFJO] 2014/02/01 07:33:03 INFO:   2559 files,   1.82 GiB in local repo
[YUFJO] 2014/02/01 07:33:03 INFO:     66 files, 213.06 MiB to synchronize
panic: runtime error: slice bounds out of range

goroutine 34 [running]:
runtime.panic(0x7561e0, 0xc3136a)
        /home/jb/go-sunos/src/pkg/runtime/panic.c:264 +0xb6
github.com/calmh/syncthing/model.(*FileQueue).Get(0xc2100e0280, 0xc21068c540, 0x34, 0x0, 0x0, ...)
        /home/jb/src/github.com/calmh/syncthing/model/filequeue.go:121 +0x48e
github.com/calmh/syncthing/model.func·004()
        /home/jb/src/github.com/calmh/syncthing/model/model.go:511 +0x45f
created by github.com/calmh/syncthing/model.(*Model).AddConnection
        /home/jb/src/github.com/calmh/syncthing/model/model.go:522 +0x266

```

**Comment by DenisCabreira** [comment]
I believe I have come across this, or a similar bug:

    [BFHOS] 17:01:05 INFO: syncthing v0.10.29 (go1.4.2 linux-386 default) unknown-user@syncthing-builder 2015-03-22 14:09:14 UTC
    [BFHOS] 17:01:05 INFO: My ID: BFHOS3U-LWKIQ5P-2S44XWQ-PBSJSL3-O7YUAW2-NMM2NEV-JGFPOO3-F64ABQQ
    [BFHOS] 17:01:06 INFO: Starting web GUI on http://127.0.0.1:8080/
    [BFHOS] 17:01:08 INFO: Starting UPnP discovery...
    [BFHOS] 17:01:11 INFO: Invalid IGD response: invalid device UUID bc4b25d-4a71-371d-bd98-038eb5924b23 (continuing anyway)
    [BFHOS] 17:01:11 INFO: Invalid IGD response: invalid device UUID bc4b25d-4a71-371d-bd98-038eb5924b23 (continuing anyway)
    [BFHOS] 17:01:11 INFO: Invalid IGD response: invalid device UUID bc4b25d-4a71-371d-bd98-038eb5924b23 (continuing anyway)
    [BFHOS] 17:01:14 INFO: UPnP discovery complete (found 3 devices).
    [BFHOS] 17:01:14 INFO: Created UPnP port mapping for external port 63512 on UPnP device 'equipo_wifimedia_R' (192.168.1.1).
    [BFHOS] 17:01:14 INFO: Starting local discovery announcements
    [BFHOS] 17:01:14 INFO: Starting global discovery announcements
    [BFHOS] 17:01:14 OK: Ready to synchronize Musica (read-write)
    [BFHOS] 17:01:14 OK: Ready to synchronize SacauntosDenis (read-write)
    [BFHOS] 17:01:14 OK: Ready to synchronize SyncDenis (read-write)
    [BFHOS] 17:01:14 INFO: Device BFHOS3U-LWKIQ5P-2S44XWQ-PBSJSL3-O7YUAW2-NMM2NEV-JGFPOO3-F64ABQQ is "SacauntosDenis" at [dynamic]
    [BFHOS] 17:01:14 INFO: Device UARDV55-OJWEADP-OKZMKPI-ZITDZQV-D4Z6KKA-ZCHGR7F-U5R5OOL-I4Q32AT is "spektr" at [dynamic]
    [BFHOS] 17:01:14 INFO: Starting usage reporting
    [BFHOS] 17:01:14 INFO: Completed initial scan (rw) of folder Musica
    [BFHOS] 17:01:14 INFO: Completed initial scan (rw) of folder SacauntosDenis
    [BFHOS] 17:01:29 INFO: Established secure connection to UARDV55-OJWEADP-OKZMKPI-ZITDZQV-D4Z6KKA-ZCHGR7F-U5R5OOL-I4Q32AT at 192.168.1.2:22000-192.168.1.207:36246
    [BFHOS] 17:01:29 INFO: Connection to UARDV55-OJWEADP-OKZMKPI-ZITDZQV-D4Z6KKA-ZCHGR7F-U5R5OOL-I4Q32AT closed: EOF
    [BFHOS] 17:01:45 INFO: Established secure connection to UARDV55-OJWEADP-OKZMKPI-ZITDZQV-D4Z6KKA-ZCHGR7F-U5R5OOL-I4Q32AT at [fe80::92e6:baff:fec4:c6a%eth0]:41166-[fe80::5a94:6bff:feaa:c038%eth0]:22000
    panic: runtime error: slice bounds out of range
    
    goroutine 222 [running]:
    github.com/calmh/xdr.(*Reader).ReadBytesMaxInto(0x1a721380, 0x40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)

**Comment by AudriusButkevicius** [comment]
Your's is most likely two incompatible versions connecting. Check that all devices are either on 0.10 or 0.11.

**Comment by DenisCabreira** [comment]
Indeed, that was the case. Thank you! :)

Shouldn't Syncthing detect incompatible versions and ignore them, perhaps after spawning a warning, but keep working with other peers that have compatible ones, rather than crashing?

**Comment by AudriusButkevicius** [comment]
It warns you when you upgrade.
If the protocol is incompatible, we might not even be able to read the version.

**Comment by calmh** [comment]
Holy issue necromancy, batman! :)

**Comment by calmh** [comment]
The actual crash is fixed though, in v0.11.
